File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Kiểm tra số hoàn hảo/SoHoanHao.cs
using System;
namespace Shh
{
    public class Program
    {
        static void Main(string[]args)
        {
            int n = int.Parse(Console.ReadLine());
            SHH(n);
        }
        static void SHH(int n)
        {
            int tong = 0;
            for(int i = 1; i < n; i++)
                if(n % i == 0)
                    tong += i;
            if(tong == n)
                Console.Write("Yes");
            else
                Console.Write("No");
        }
    }
}

File: Upcoder/Lập trình cơ bản/3.Vòng lặp/Kiểm tra số hoàn hảo/SoHoanHao.cpp
#include<bits/stdc++.h>
using namespace std;
bool SHH(int n)
{
    int tong = 0;
    for(int i = 1; i < n; i++)
        if(n % i == 0)
            tong+= i;
    if(tong == n)
        return true;
    return false;
}
int main()
{
    int n;
    cin >> n;
    if(SHH(n))
        cout << "Yes";
    else
        cout << "No";
    return 0;
}